.github/workflows/linting.yaml

```diff
@@ -0,0 +1,32 @@
+name: "Linting"
+
+on:
+  push:
+    paths:
+      - './**'
+      - '.github/workflows/**'
+  pull_request:
+    branches: [ dev, main ]
+    paths:
+      - './**'
+
+jobs:
+  # Non-blocking job that checks .md file locations and also lints them
+  markdown:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the code
+        uses: actions/checkout@v2
+      - name: "Validate markdown file locations"
+        continue-on-error: true
+        run: "scripts/check-markdown-files.sh" 
+      - name: markdownlint-cli2-action
+        uses: DavidAnson/markdownlint-cli2-action@v9
```

Comment:
I'm sure there's a reasoned walkthrough of this but as someone who hasn't dug deep into the world of markdown lint actions out there - why this particular one? Do we trust the source?

Reply:
Yes, David Ansons's `markdownlint` implementation is the generally accepted standard for markdown linting. It's also the recommended linter GSF included in [ADR-0004](https://github.com/Green-Software-Foundation/carbon-aware-sdk/blob/dev/docs/architecture/decisions/0004-documentation.md#decision) for linting the readmes in this repo. 

We could write a script that calls the `markdownlint-cli` individually and does all the steps to turn it into the action but the `GithubAction` implementation is included in the standard library of implementations they provide (nuget package, CLI, github action, vscode extension, ESLint extension etc.) so I trust it.

Reply:
Given that this is the one discussed in that ADR, I got it and it makes sense to me. Thanks for the info!